contracts/BancorAdaptor.sol

```diff
@@ -0,0 +1,102 @@
+pragma solidity ^0.4.22;
+
+import "./Marketplace.sol";
+
+
+
+
+contract IERC20Token {
+	// these functions aren't abstract since the compiler emits automatically generated getter functions as external
+	function name() public view returns (string) {}
+	function symbol() public view returns (string) {}
+	function decimals() public view returns (uint8) {}
+	function totalSupply() public view returns (uint256) {}
+	function balanceOf(address _owner) public view returns (uint256) { _owner; }
+	function allowance(address _owner, address _spender) public view returns (uint256) { _owner; _spender; }
+
+	function transfer(address _to, uint256 _value) public returns (bool success);
+	function transferFrom(address _from, address _to, uint256 _value) public returns (bool success);
+	function approve(address _spender, uint256 _value) public returns (bool success);
+}
+
+contract IBancorConverter {
+	function quickConvert(IERC20Token[] _path, uint256 _amount, uint256 _minReturn) public payable returns (uint256) {}
+}
+
+
+
+contract BancorAdaptor {
+    using SafeMath for uint256;
+
+	address public marketplace_address;
+	address public bancor_converter_address;
+	address public datacoin_address;
+	constructor(address _marketplace_address, address _bancor_converter_address, address _datacoin_address) public {
+		marketplace_address = _marketplace_address;
+		bancor_converter_address = _bancor_converter_address;
+		datacoin_address = _datacoin_address;
+	}
+/*
+return price per second of product in DATA, or reverts() if none found
+*/
+	function getProductPrice(Marketplace mkt, bytes32 productId) internal returns (uint) {
+		(string memory name, address owner, address beneficiary, uint pricePerSecond, Marketplace.Currency priceCurrency, uint minimumSubscriptionSeconds, Marketplace.ProductState state) = mkt.getProduct(productId);
+		require(owner != 0x0, "not found");
+		require(priceCurrency == Marketplace.Currency.DATA, "price not in DATA");
```

Comment:
`priceCurrency` defines what currency the price is defined in (the options are DATA and USD). The payment always happens in DATA, so we should support both price definitions. The Marketplace contract contains conversion rate (and maybe a helper function too) to convert a price defined in USD to DATA. `getProductPrice` here could always return the price in DATA.

Reply:
ok, I'll change

Reply:
fixed in latest commit